sway-book/src/documentation/misc/advanced-concepts/structs.md

```diff
@@ -0,0 +1,5 @@
+# Struct Memory Layout
```

Comment:
Structs and enums are mentioned here and in "built in types". we probably only need one section